controllers/storagecluster/cephcluster.go

```diff
@@ -98,11 +98,14 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 	if sc.Spec.ExternalStorage.Enable {
 		cephCluster = newExternalCephCluster(sc, r.images.Ceph, r.monitoringIP, r.monitoringPort)
 	} else {
-		kmsConfigMap, err := getKMSConfigMap(sc, r.Client, reachKMSProvider)
+		kmsConfigMap, err := getKMSConfigMap(sc, r.Client)
 		if err != nil {
 			r.Log.Error(err, "failed to procure KMS config")
 			return err
 		}
+		if err = reachKMSProvider(kmsConfigMap); err != nil {
```

Comment:
It seems that in your refactor, you are now trying to run `reachKMSProvider()` on every reconcile, whereas before you only ran it if `kmsConfigMap != nil`. I don't fully understand in what scenario it's okay to have a `nil` ConfigMap pointer with also having returned an error, but this seems to be what's causing your unit test failures.

Reply:
True, I missed that check. Added the check and TODO comment. @jarrpa , please take a look

Reply:
:+1: